File: tests/test_highlighter.c
#include "acutest.h"
#include "../src/core/highlighter.h"
#include "../src/core/theme.h"
#include <string.h>

void test_highlighter_basic(void) {
    const Theme *theme = theme_default();
    char out[1024];
    
    // Keyword
    highlighter_highlight("if", theme, out, sizeof(out));
    TEST_CHECK(strstr(out, "if") != NULL);
    TEST_CHECK(strstr(out, "<span foreground=") != NULL);

    // Comment
    highlighter_highlight("// comment", theme, out, sizeof(out));
    TEST_CHECK(strstr(out, "comment") != NULL);
    TEST_CHECK(strstr(out, "<span foreground=") != NULL);

    // Symbols
    highlighter_highlight("{}", theme, out, sizeof(out));
    TEST_CHECK(strstr(out, "{") != NULL);
    TEST_CHECK(strstr(out, "}") != NULL);
    TEST_CHECK(strstr(out, "<span foreground=") != NULL);
}

void test_highlighter_escaping(void) {
    const Theme *theme = theme_default();
    char out[1024];
    
    // Escaping
    highlighter_highlight("a < b && c > d", theme, out, sizeof(out));
    TEST_CHECK(strstr(out, "&lt;") != NULL);
    TEST_CHECK(strstr(out, "&gt;") != NULL);
    TEST_CHECK(strstr(out, "&amp;") != NULL);
}

void test_highlighter_string(void) {
    const Theme *theme = theme_default();
    char out[1024];
    
    // String
    highlighter_highlight("\"hello world\"", theme, out, sizeof(out));
    TEST_CHECK(strstr(out, "hello world") != NULL);
    TEST_CHECK(strstr(out, "<span foreground=") != NULL);
}

TEST_LIST = {
    { "highlighter_basic", test_highlighter_basic },
    { "highlighter_escaping", test_highlighter_escaping },
    { "highlighter_string", test_highlighter_string },
    { NULL, NULL }
};

File: python/test_highlighter.py
import unittest
import os
import sys
from c_slides_bindings import CSlides

class TestHighlighter(unittest.TestCase):
    @classmethod
    def setUpClass(cls):
        # Asumimos que libslider.so está en la raíz del proyecto
        lib_path = os.path.abspath(os.path.join(os.path.dirname(__file__), "..", "libslider.so"))
        cls.cs = CSlides(lib_path)
        cls.theme = cls.cs.theme_default()

    def test_highlighter_basic(self):
        # Keyword
        out = self.cs.highlight("if", self.theme)
        self.assertIn("if", out)
        self.assertIn("<span foreground=", out)

        # Comment
        out = self.cs.highlight("// comment", self.theme)
        self.assertIn("comment", out)
        self.assertIn("<span foreground=", out)

        # Symbols
        out = self.cs.highlight("{}", self.theme)
        self.assertIn("{", out)
        self.assertIn("}", out)
        self.assertIn("<span foreground=", out)

    def test_highlighter_escaping(self):
        # Escaping
        out = self.cs.highlight("a < b && c > d", self.theme)
        self.assertIn("&lt;", out)
        self.assertIn("&gt;", out)
        self.assertIn("&amp;", out)

    def test_highlighter_string(self):
        # String
        out = self.cs.highlight("\"hello world\"", self.theme)
        self.assertIn("hello world", out)
        self.assertIn("<span foreground=", out)

if __name__ == "__main__":
    unittest.main()
